File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/mad/Java/C++/s831281137_commented.cpp
#include <iostream>
#include <vector>
#include <algorithm>

int main() {
    int N, W;
    std::cin >> N >> W;

    std::vector<int> v(N + 1);
    std::vector<int> w(N + 1);

    for (int i = 1; i <= N; i++) {
        std::cin >> v[i] >> w[i];
    }

    std::vector<std::vector<int>> sumV(N + 1, std::vector<int>(W + 1, 0));

    for (int i = 1; i <= N; i++) {
        for (int i_sum = 0; i_sum <= W; i_sum++) {
            if (i_sum < w[i]) {
                sumV[i][i_sum] = sumV[i - 1][i_sum];
            } else {
                sumV[i][i_sum] = std::max(sumV[i - 1][i_sum], sumV[i - 1][i_sum - w[i]] + v[i]);
            }
        }
    }

    std::cout << sumV[N][W] << std::endl;
}

// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/Java/Python/s831281137_commented.py
import sys

def input_list():
    return list(map(int, sys.stdin.read().split()))

data = input_list()
ptr = 0

def nint():
    global ptr
    ret = data[ptr]
    ptr += 1
    return ret

def main():
    N = nint()
    W = nint()
    v = [0] * (N + 1)
    w = [0] * (N + 1)
    for i in range(1, N+1):
        v[i] = nint()
        w[i] = nint()
    sumV = [[0]*(W+1) for _ in range(N+1)]
    for i in range(1, N+1):
        for i_sum in range(w[i]):
            sumV[i][i_sum] = sumV[i-1][i_sum]
        for i_sum in range(w[i], W+1):
            sumV[i][i_sum] = max(sumV[i-1][i_sum], sumV[i-1][i_sum - w[i]] + v[i])
    print(sumV[N][W])

if __name__ == "__main__":
    main()  # End of Code
